test/parse/test.sh

```
@@ -67,10 +67,18 @@ make
 for file in "${!test_cases[@]}"; do
     expected_status="${test_cases[$file]}"
     ./parse "$file" 2> /dev/null
-    if [ "$?" -eq "$expected_status" ]; then
-        echo -e "$file: \e[32mOK\e[0m"
+    if [ "$expected_status" -eq 0 ]; then
+        if [ "$?" -eq 0 ]; then
```

Shellcheck (0.10.0)

[warning] This $? refers to a condition, not a command. Assign to a variable to avoid it being overwritten.

(SC2319)